rust-runtime/aws-smithy-runtime-api/src/config_bag.rs

```diff
@@ -59,42 +61,196 @@ impl Deref for FrozenConfigBag {
     }
 }
 
-pub trait Persist {
-    fn layer_name(&self) -> &'static str;
-    fn persist(&self, layer: &mut ConfigBag);
+pub(crate) mod value {
```

Comment:
What is the purpose of the module?

Reply:
keep value private while allowing it to be exposed in the trait

Reply:
Can you add a comment?